app/models/statement_row.rb

```diff
@@ -6,7 +6,15 @@ class StatementRow < ActiveRecord::Base
   validates_presence_of :order_detail_id, :statement_id
 
   before_destroy { @parent_statement = statement }
-  after_destroy { @parent_statement.destroy if @parent_statement.statement_rows.reload.empty? }
+  
+  after_destroy do
+    if @parent_statement.statement_rows.reload.empty?
+      @parent_statement.order_details.each do |order_detail|
+        order_detail.update_attributes(statement: nil)
+      end
+      @parent_statement.destroy
+    end
+  end
```

Comment:
I'm note quite sure if we care about callbacks/validations, but we could possibly get away with `@parent_statement.order_details.update_all(statement: nil)`. If we care about `updated_at`, we could include that in there, too.

Another option I don't know if it would work is to add `dependent: ::nullify` on the Statement `has_many :order_details`.